src/gpio.c

```diff
@@ -0,0 +1,147 @@
+#include "gpio.h"
+
+// Includes
+#include <zephyr/logging/log.h>
+
+// Definitions
+
+// Protototypes
+
+// Globals
+
+static const struct gpio_dt_spec led0 = GPIO_DT_SPEC_GET(LED0_NODE, gpios);
+static const struct gpio_dt_spec led1 = GPIO_DT_SPEC_GET(LED1_NODE, gpios);
+static const struct gpio_dt_spec led2 = GPIO_DT_SPEC_GET(LED2_NODE, gpios);
+static const struct gpio_dt_spec led3 = GPIO_DT_SPEC_GET(LED3_NODE, gpios);
+
+// Functions
+LOG_MODULE_REGISTER(gpio, CONFIG_GPIO_LOG_LEVEL);
+
+void otLedSetOff(uint8_t aLed) {
+    int err = 0;
+	switch (aLed) {
+        case 1:
+            err = gpio_pin_configure_dt(&led0, GPIO_OUTPUT_INACTIVE);
+            break;
+        case 2:
+            err = gpio_pin_configure_dt(&led1, GPIO_OUTPUT_INACTIVE);
+            break;
+        case 3:
+            err = gpio_pin_configure_dt(&led2, GPIO_OUTPUT_INACTIVE);
+            break;
+        case 4:
+            err = gpio_pin_configure_dt(&led3, GPIO_OUTPUT_INACTIVE);
+            break;
+    }
+	if (err < 0) {
+		LOG_WRN("LED set OFF failed");
+	}
+
+}
+
+void otLedSetOn(uint8_t aLed) {
+	int err = 0;
+	switch (aLed) {
+        case 1:
+            err = gpio_pin_configure_dt(&led0, GPIO_OUTPUT_ACTIVE);
+            break;
+        case 2:
+            err = gpio_pin_configure_dt(&led1, GPIO_OUTPUT_ACTIVE);
+            break;
+        case 3:
+            err = gpio_pin_configure_dt(&led2, GPIO_OUTPUT_ACTIVE);
+            break;
+        case 4:
+            err = gpio_pin_configure_dt(&led3, GPIO_OUTPUT_ACTIVE);
+            break;
+    }
+	if (err < 0) {
+		LOG_WRN("LED set ON failed");
+	}
+}
+
+void otLedToggle(uint8_t aLed) {
+	int err = 0;
+	switch (aLed) {
+        case 1:
+            err = gpio_pin_toggle_dt(&led0);
+            break;
+        case 2:
+            err = gpio_pin_toggle_dt(&led1);
+            break;
+        case 3:
+            err = gpio_pin_toggle_dt(&led2);
+            break;
+        case 4:
+            err = gpio_pin_toggle_dt(&led3);
+            break;
+    }
+	if (err < 0) {
+		LOG_WRN("LED toggle failed");
+	}
+}
+
+void otLedBlink(uint8_t aLed) {
+    switch (aLed) {
+        case 1:
+            otLedToggle(1);
```

Comment:
I don't think we should have sleeps in these functions. Sleeps can block important things from happening so let's not to this. Probably best taking the function out if it is blocking

Reply:
I'll remove the Blink function!